### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by quliyevrustam**
> Create a new module - Bulbulatory_Recommendations. The module must have a configuration in admin panel that will allow defining if the functionality is enabled or disabled.

DoneI

- module Bulbulatory_Recommendations created
- module - Bulbulatory_Recommendations have a configuration in admin panel

### CODE REVIEW

File: app/code/Bulbulatory/Recommendations/etc/config.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Store:etc/config.xsd">
+    <default>
+        <recommendations>
+            <general>
+                <enable>1</enable>
```
Comment (rafdol) [inline app/code/Bulbulatory/Recommendations/etc/config.xml]: It is better if feature is disabled by default (client decide when turn it on).

File: app/code/Bulbulatory/Recommendations/etc/adminhtml/system.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            <label>Bulbulatory</label>
+        </tab>
+        <section id="recommendations" translate="label" type="text" sortOrder="130" showInDefault="1" showInWebsite="1" showInStore="1">
+            <class>separator-top</class>
+            <label>Recommendations</label>
+            <tab>bulbulatory</tab>
+            <resource>Bulbulatory_Recommendations::recommendations_config</resource>
+            <group id="general" translate="label" type="text" sortOrder="10" showInDefault="1" showInWebsite="0" showInStore="0">
+                <label>General Configuration</label>
+                <field id="enable" translate="label" type="select" sortOrder="1" showInDefault="1" showInWebsite="0" showInStore="0">
```
Comment (rafdol) [inline app/code/Bulbulatory/Recommendations/etc/adminhtml/system.xml]: Try to using sort order values like 10, 20, 30, etc. (then you can in fast way put new field without changing others sort order values).

File: app/code/Bulbulatory/Recommendations/etc/module.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Bulbulatory_Recommendations" setup_version="0.0.1">
```
Comment (rafdol) [inline app/code/Bulbulatory/Recommendations/etc/module.xml]: I suggest to use the last part of the version number for bug fixing only. You can read more about versioning in this article: https://en.wikipedia.org/wiki/Software_versioning.

File: app/code/Bulbulatory/Recommendations/etc/adminhtml/system.xml (hunk 2 above)
Comment (quliyevrustam) [reply app/code/Bulbulatory/Recommendations/etc/adminhtml/system.xml]: @rafdol 

> Try to using sort order values like 10, 20, 30, etc. (then you can in fast way put new field without changing others sort order values).

sortOrder numbering fixed!

File: app/code/Bulbulatory/Recommendations/etc/config.xml (hunk 1 above)
Comment (quliyevrustam) [reply app/code/Bulbulatory/Recommendations/etc/config.xml]: @rafdol 

> It is better if feature is disabled by default (client decide when turn it on).

I made "disabled" by default!

File: app/code/Bulbulatory/Recommendations/etc/module.xml (hunk 3 above)
Comment (quliyevrustam) [reply app/code/Bulbulatory/Recommendations/etc/module.xml]: @rafdol 

> I suggest to use the last part of the version number for bug fixing only. You can read more about versioning in this article: https://en.wikipedia.org/wiki/Software_versioning.

Thanks for the advice and link. I fixed module versioning. I made version of module - 1.0.1

1 - show current version
0 - show some changes in current version
1 - show my bugfix, that I made today

**Pull request merged**